src/content/docs/browser/new-relic-browser/configuration/proxy-agent-requests.mdx

```diff
@@ -0,0 +1,82 @@
+---
+title: Proxy agent chunk requests and analytics data
+tags:
+  - Browser monitoring
+  - Proxy
+metaDescription: "The agent can be configured to fetch its code and send analytics through a proxy server instead of New Relic's endpoints."
+---
+
+From [version 1.240.0](/docs/release-notes/new-relic-browser-release-notes/browser-agent-release-notes) and higher, the browser agent officially allows a proxy url to be set when determining either or both of:
+
+1. Where to fetch or lazy-load agent code.
+2. Where to send analytics.
+
+Today, many ad-blockers and security VPNs block client requests to third-party domains including New Relic's. As the number of internet users opting into the use of ad-blockers increase, customers will continue to lose observability on those devices. One solution would be to proxy those requests though an unblocked middleman. For example, `yoursite.com` can send its visitor analytics through its `nrbeacon.yoursite.com` proxy which would not be blocked by anti-adware being that it's considered a first-party (sub)domain.
+
+<Callout variant="important">
+  This feature is currently available for those using the copy-and-paste _or_ NPM installation methods. There is currently no UI or NerdGraph configuration options available. We are continuing to work on improving access to these and other configuration options.
+</Callout>
+
+## How to use a proxy [#use]
+
+The agent utilizes two configuration properties when determining where to fetch chunks or send data.
+
+For example, using copy and paste:
+
+```js
+window.NREUM.init = {
+    ...<other init properties>,
+    proxy: {
+        assets: <URL to fetch agent chunks>,
+        beacon: <URL to send analytics>
+    }
+}
+```
+
+For NPM, similarly just pass the `init` object with a proxy as part of any other configurations:
+
+```js
+new BrowserAgent({ init: { proxy: <object similar to above> } })
+```
+
+`init.proxy.assets` is used verbatim in the copy-and-paste installation to fetch agent chunks.
+
+* For example, the v1.240.0 Pro+SPA loader will require `nr-spa.___-1.240.min.js` file.
+* If this field is set to be `nrassets.abc.com` reverse proxy, provide this value as the exact URL, including the scheme and ending slash: `'https://nrassets.abc.com/'`.
```

Comment:
```suggestion
* If this field is set to be `nrassets.abc.com`, the agent code will be loaded from `https://nrassets.abc.com/nr-spa.<hash>-1.240.0.min.js` instead of `https://js-agent.newrelic.com/nr-spa.<hash>-1.240.0.min.js`
```

Reply:
Could you revise this edit; the message is lost here. It's important that they explicitly use `https://` and `/` at the end when providing the value for this property.

Reply:
I added the `https://` piece but our code assumes the provided value does not end in a slash.

Reply:
it's different for the asset path vs beacon:
<img width="618" alt="image" src="https://github.com/newrelic/docs-website/assets/28714891/98304ebf-a046-49e9-a395-43588e25dc11">
Webpack will just prepend that new url as is, so the ending slash must be specified.

Reply:
Or it's going to end up fetching from `https://nrassets.abc.comnr-spa.<hash>-1.240.0.min.js`

Reply:
This is in the agent right now:
<img width="1275" alt="image" src="https://github.com/newrelic/docs-website/assets/3984979/4d2628cf-5744-4608-8323-8d86d8673ea3">